src/common/lib/client/realtimepresence.ts

```diff
@@ -446,7 +438,15 @@ class RealtimePresence extends EventEmitter {
       );
       // RTP17g: Send ENTER containing the member id, clientId and data
       // attributes.
-      Utils.whenPromiseSettles(this._enterOrUpdateClient(entry.id, entry.clientId, entry.data, 'enter'), reenterCb);
+      // RTP17g1: suppress id if the connId has changed
+      const id = (entry.connectionId === connId) ? entry.id : undefined;
+      this._enterOrUpdateClient(id, entry.clientId, entry.data, 'enter').catch((err) => {
+        const msg = 'Presence auto-re-enter failed: ' + err.toString();
+        const wrappedErr = new ErrorInfo(msg, 91004, 400);
```

Comment:
consider using `ErrorInfo.cause` instead of concatenating error messages:
```suggestion
        const wrappedErr = new ErrorInfo('Presence auto re-enter failed', 91004, 400, err);
```

Reply:
sure why not

Reply:
although I still need to concatenate it for the error log, ably-js doesn't have any structured logging, the log line is just a string